validphys2/src/validphys/pseudodata.py

```diff
@@ -345,29 +385,52 @@ def make_level1_data(
     >>> l1_cd
     [CommonData(setname='NMC', ndata=204, commondataproc='DIS_NCE', nkin=3, nsys=16)]
     """
-
+    # =============== generate experimental covariance matrix ===============#
     dataset_input_list = list(data.dsinputs)
+    original_level0_commondata_wc = copy.deepcopy(level0_commondata_wc)
+
+    if type1_inconsistency:
+        level0_commondata_wc = [
+                                InconsistentCommonData(setname=cd.setname, ndata=cd.ndata, 
+                                                    commondataproc=cd.commondataproc, 
+                                                    nkin=cd.nkin, nsys=cd.nsys, 
+                                                    commondata_table = cd.commondata_table, 
+                                                    systype_table = cd.systype_table) 
+                                for cd in level0_commondata_wc
+                        ]
+
+        level0_commondata_wc = [cd.process_commondata(ADD,MULT,CORR,UNCORR,SPECIAL,inconsistent_datasets,sys_rescaling_factor)
+                            for cd in level0_commondata_wc]
 
     covmat = dataset_inputs_covmat_from_systematics(
-        level0_commondata_wc,
+        level0_commondata_wc, 
         dataset_input_list,
         use_weights_in_covmat=False,
         norm_threshold=None,
         _list_of_central_values=None,
         _only_additive=sep_mult,
     )
 
-    # ================== generation of Level1 data ======================#
+    # ================== generation of pseudo data ======================#
+    # generate pseudo data starting from theory predictions
+    # covmat can be an inconsistent one
+
     level1_data = make_replica(
         level0_commondata_wc, filterseed, covmat, sep_mult=sep_mult, genrep=True
     )
 
     indexed_level1_data = indexed_make_replica(experiments_index, level1_data)
 
-    # ===== create commondata instances with central values given by pseudo_data =====#
-    level1_commondata_dict = {c.setname: c for c in level0_commondata_wc}
+    # for an inconsistent fit use commondata with unmodified systematics and only change central value
+    if not reference_fit:
+        level1_commondata_dict = {c.setname: c for c in original_level0_commondata_wc}
+
+    # for a consistent fit use commondata with modified systematics and also change central value
+    else:
+        level1_commondata_dict = {c.setname: c for c in level0_commondata_wc}
```

Comment:
Basically the only (new) information that needs to be passed to this function is whether there exist an original level 0 common data.

So, in my view, the data should be prepared before calling `make_level1_pseudodata` and the only argument that needs to be added is `original_level0_commondata_wc=None`.

Then you can do e.g.

```python
if original_level_0_commondata_wc is None:
    level1_commondata_dict = {c.setname: c for c in level0_commondata_wc}
else:
    level1_commondata_dict = {c.setname: c for c in original_level0_commondata_wc}
```